src/main/java/org/graylog2/gelfclient/transport/GelfSenderThread.java

```diff
@@ -139,4 +146,40 @@ public void stop() {
         keepRunning.set(false);
         senderThread.interrupt();
     }
+
+    /**
+     * Block and wait for all messages in the queue to send. This can be used during shutdown to wait for
+     * queued messages to send before shutting down.
+     * Waiting will continue for the indicated {@code FLUSH_WAIT_RETRIES} and {@code FLUSH_WAIT_MS.}
+     */
+    void flushSynchronously() {
```

Comment:
@bernd The operative part of this change is this new `flushSynchronously()` method on the `GelfSenderThreat` that blocks and waits for queued and in-flight messages to flush/send. This is called from a new `flushAndStop()` method in `GelfTCPTransport` and `GelfUDPTransport`.

Does this approach look ok?

Reply:
I really like this, and it will simplify the looping logic. Will do.